File: src/main/java/com/procurement/access/model/dto/ocds/Renewal.java
package com.procurement.access.model.dto.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import javax.validation.constraints.NotNull;
import lombok.Getter;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

@Getter
public class Renewal {

    @NotNull
    @JsonProperty("hasRenewals")
    private final Boolean hasRenewals;

    @JsonCreator
    public Renewal(@JsonProperty("hasRenewals") final Boolean hasRenewals) {
        this.hasRenewals = hasRenewals;
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder()
                .append(hasRenewals)
                .toHashCode();
    }

    @Override
    public boolean equals(final Object other) {
        if (other == this) {
            return true;
        }
        if (!(other instanceof Renewal)) {
            return false;
        }
        final Renewal rhs = (Renewal) other;
        return new EqualsBuilder()
                .append(hasRenewals, rhs.hasRenewals)
                .isEquals();
    }
}

File: src/main/kotlin/com/procurement/access/model/dto/ocds/Renewal.kt
package com.procurement.access.model.dto.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.procurement.access.model.dto.databinding.BooleansDeserializer
import javax.validation.constraints.NotNull

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Renewal @JsonCreator constructor(

        @field:NotNull
        @field:JsonDeserialize(using = BooleansDeserializer::class)
        @get:JsonProperty("hasRenewals")
        val hasRenewals: Boolean?
)
